src/laddie/laddie_main.f90

```diff
@@ -90,35 +136,399 @@ SUBROUTINE initialise_laddie_model( mesh, laddie, ocean, ice)
 
     ! Allocate variables
     CALL allocate_laddie_model( mesh, laddie)
+    CALL allocate_laddie_timestep( mesh, laddie%now)
+
+    ! Allocate timestep
+    SELECT CASE(C%choice_laddie_integration_scheme)
+      CASE DEFAULT
+        CALL crash('unknown choice_laddie_integration_scheme "' // TRIM( C%choice_laddie_integration_scheme) // '"')
+      CASE ('euler')
+        CALL allocate_laddie_timestep( mesh, laddie%np1)
+      CASE ('fbrk3')
+        CALL allocate_laddie_timestep( mesh, laddie%np13)
+        CALL allocate_laddie_timestep( mesh, laddie%np12)
+        CALL allocate_laddie_timestep( mesh, laddie%np1)
+    END SELECT
+
+    ! Mask on a grid
+    DO vi = mesh%vi1, mesh%vi2
+      laddie%mask_a( vi)  = ice%mask_floating_ice( vi)
+    END DO
 
     ! Layer thickness 
     DO vi = mesh%vi1, mesh%vi2
-       IF (ice%mask_floating_ice( vi)) THEN
-         laddie%H( vi)      = C%laddie_initial_thickness
-         laddie%H_prev( vi) = C%laddie_initial_thickness
-         laddie%H_next( vi) = C%laddie_initial_thickness
+       IF (laddie%mask_a( vi)) THEN
+         laddie%now%H( vi)      = C%laddie_initial_thickness
+         SELECT CASE(C%choice_laddie_integration_scheme)
```

Comment:
Good point. This is initialisation, so has a minimal impact on total runtime. But it's good practice anyway.

In this specific case, I'll create two separate routines, see 2 comments down. So that resolves this issue.